### PR TITLE
#26 Refactor form-handling code

### PR DESCRIPTION
**Pull request by liza-veis**

### CODE REVIEW

**Review by R-Mielamud — APPROVED** [review]

**Review by katyakostikova — APPROVED** [review]

File: src/components/form/form-input/form-input.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import { View } from 'react-native';
+import { FormikValues, useFormikContext } from 'formik';
+
+import { Input, Text } from 'src/components';
+import styles from './styles';
+
+type InputProps = React.ComponentPropsWithoutRef<typeof Input>;
+
+type FormInputProps = Omit<InputProps, 'value' | 'onChangeText' | 'onBlur'> & {
+  name: string;
```
Comment (AleksandrTermenzhy) [inline src/components/form/form-input/form-input.tsx:11]:
```suggestion
  name: string;
  containerStyle?: StyleProp<ViewStyle>;
```

File: src/components/form/form-input/form-input.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+};
+
+const FormInput: React.FC<FormInputProps> = ({ name, ...inputProps }) => {
```
Comment (AleksandrTermenzhy) [inline src/components/form/form-input/form-input.tsx]:
```suggestion
const FormInput: React.FC<FormInputProps> = ({ name, containerStyle, ...inputProps }) => {
```

File: src/components/form/form-input/form-input.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  const { values, errors, touched, handleChange, handleBlur } =
+    useFormikContext<FormikValues>();
+
+  const error = touched[name] && errors[name];
+
+  return (
+    <View style={styles.field}>
```
Comment (AleksandrTermenzhy) [inline src/components/form/form-input/form-input.tsx]:
```suggestion
    <View style={[styles.field, containerStyle]}>
```

File: src/components/form/form-input/styles.ts [hunk 4]
```diff
@@ -1,15 +1,16 @@
...
 const styles = StyleSheet.create({
-  form: {
-    height: '100%',
-    paddingVertical: 50,
-    paddingHorizontal: 20,
-    alignItems: 'center',
-  },
   field: {
+    width: '100%',
     paddingBottom: 15,
```
Comment (AleksandrTermenzhy) [inline src/components/form/form-input/styles.ts]:
```suggestion
```

File: src/components/form/form-input/styles.ts [hunk 5]
```diff
@@ -1,15 +1,16 @@
...
+
 const styles = StyleSheet.create({
-  form: {
-    height: '100%',
-    paddingVertical: 50,
-    paddingHorizontal: 20,
-    alignItems: 'center',
-  },
   field: {
+    width: '100%',
```
Comment (AleksandrTermenzhy) [inline src/components/form/form-input/styles.ts]: Do we really nee it here?

File: src/components/form/form-input/styles.ts [hunk 6]
```diff
@@ -1,15 +1,16 @@
...
     paddingBottom: 15,
   },
+  error: {
+    paddingTop: 4,
+    color: AppColor.ACCENT,
```
Comment (AleksandrTermenzhy) [inline src/components/form/form-input/styles.ts]: I'd change it from accent to more explicit variant of red color

**Review by AleksandrTermenzhy — APPROVED** [review]

**Pull request merged**